turbinia/evidence.py

```diff
@@ -355,3 +356,44 @@ def __init__(self, artifact_name):
     super(ExportedFileArtifact, self).__init__()
     self.artifact_name = artifact_name
     self.copyable = True
+
+
+class DockerContainer(Evidence):
+  """Evidence object for a DockerContainer filesystem.
+
+  Attributes:
+    container_id(str): The ID of the container to mount.
+  """
+
+  def __init__(self, container_id=None, *args, **kwargs):
+    """Initialization for Docker Container."""
+    super(DockerContainer, self).__init__(*args, **kwargs)
+    self.container_id = container_id
+    self._container_fs_path = None
+    self._docker_root_directory = None
+    self._mount_path = None
+    self.context_dependent = True
+
+  def _preprocess(self):
+    self._mount_path = self.parent_evidence.local_path
+    if issubclass(type(self.parent_evidence), RawDisk):
```

Comment:
TODO'ed
Though I think there may be a case where, for example, plaso will be able to do things on a disk, while we won't be able to mount its partitions (ie; 'dirty' FS that the kernel won't know how to mount, etc)

Reply:
Interesting.  Ok, SG.

Reply:
Thinking about this again, I really think we should try to put this into the RawDisk preprocessor, but if it can't mount any partitions, then we can just emit a warning and for those tasks that require a mounted partition we can have a check for that.  I think the longer term solution might be to have a separate evidence type for RawDiskPartition, and those get enumerated out by a separate task.  For now we can keep this as a TODO here since the other processors do similar things (e.g. GoogleCloudDisk).  I think we should move the mount that happens in the task run code though, so I'll comment on that below.

Reply:
done